doc/paper/paper.tex

```diff
@@ -0,0 +1,1012 @@
+  \newcommand{\q}[1]{``#1''}
+
+  \title{dp3: multimodal log database}
+  \author{
+    Wyatt Alt
+    \texttt{wyatt.alt@gmail.com}
+    \and
+    Jainil Ajmera
+    \texttt{jainil@berkeley.edu}
+    }
+
+    \begin{document}
+    \maketitle
+
+    \begin{abstract}
+      dp3 is a multimodal log database for robotics data infrastructure, with goals
+      to support low-latency playback, timeseries analytics, precomputed
+      multigranular field-level statistical summaries, and efficient integration with
+      distributed compute under a single system.
+
+      The first section of this paper will introduce key concepts, background, and
+      problems to solve in the multimodal logging domain. The second section will
+      present the architecture of dp3.
+    \end{abstract}
+
+    \section{Introduction}
+    dp3 is a row-oriented multimodal log database structured around a
+    time-partitioned copy-on-write tree, which is stored in S3-compatible
+    storage and fronted by an in-memory cache of inner tree nodes. The tree
+    structure is heavily inspired by btrdb \cite{Andersen}. It is designed for
+    storage and retrieval of robotics data with three \q{user personas} in
+    mind:
+
+    The \textbf{robotics developer}, whose primary workload consists of
+    alternating between searches for interesting log events and doing replay-based
+    analysis of those events using a tool like Foxglove \cite{foxglove}, or
+    possibly replaying logs through their software stack.
+
+    The \textbf{ML developer}, whose primary workload consists of large-scale compute
+    jobs against narrow selections of topics.
+
+    The \textbf{data infrastructure admin}, who is responsible for administering the
+    database and satisfying the other two customers in a cost-effective and
+    maintainable manner.
+
+    One person may of course assume multiple personas.
+
+    For the robotics developer, dp3 provides low-latency playback for data recorded
+    by a \q{producer} from a point in time on a selection of \q{topics}, as well as a
+    query language for ergonomic expression of the \q{as-of join} type conditions
+    allowing the user to quickly pivot between searching for interesting events,
+    and replaying those events in full context.
+
+    For the ML developer, dp3 supports both client/server and embedded modes of
+    operation. The query executor may be embedded directly into jobs so that jobs
+    can operate directly against S3. This isolates heavy and bursty ML-related read
+    workloads from the comparatively lighter-weight streaming playback and
+    summarization workloads supported by the database server, and reduces the cost
+    and complexity of infrastructure management by eliminating the need to scale a
+    database deployment in step with the job workloads.
+
+    Operational burden for the data infrastructure admin is minimized by offloading
```

Comment:
For consistency, reword to "For the data infrastructure admin, operational burden is minmized / dp3 minimizes the operational burden..."?